packages/react-core/src/components/CalendarMonth/CalendarMonth.tsx

```diff
@@ -217,6 +222,31 @@ export const CalendarMonth = ({
   const changeMonth = (newMonth: number, newYear?: number) =>
     new Date(newYear ?? focusedDate.getFullYear(), newMonth, 1);
 
+  const MIN_YEAR = 1900;
+  const MAX_YEAR = 2100;
+
+  const handleYearInputChange = (event: React.FormEvent<HTMLInputElement>) => {
+    const yearStr = event.currentTarget.value;
+
+    if (/^\d{0,4}$/.test(yearStr)) {
+      setYearInput(yearStr);
+
+      if (yearStr.length === 4) {
+        const yearNum = Number(yearStr);
+        if (yearNum >= MIN_YEAR && yearNum <= MAX_YEAR) {
+          const newDate = changeYear(yearNum);
+          setFocusedDate(newDate);
+          setHoveredDate(newDate);
+          setShouldFocus(false);
```

Comment:
We need to keep the previous blur call in for FireFox. Without it if a day is selected then the year input scrolls arrows are clicked, pressing Tab acts as though focus is still on the day button rather than the year input.

```suggestion
          setShouldFocus(false);
          // We need to unfocus a date when changing year via up/down arrows in FireFox, otherwise focus doesn't go to the year input like expected
          focusRef.current?.blur(); 
```